### DISCUSSION TITLE
Why is a merge commit not used when a user retries a workflow?

### DISCUSSION THREAD

**Discussion by alexey-igrychev** [community/community#26487] [Actions]
<p><strong>During the first run</strong></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/e1d3f50f2404ec5727eb485183ba830b52e43a66.png" data-download-href="/uploads/short-url/wdLxVoXEwLSg8JIdVliqp9iXQSq.png?dl=1" title="Screenshot_16.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/17360817/181097993-381471a7-ec62-4b3f-b5c2-b4933485bc7c.png" alt="Screenshot_16.png" data-base62-sha1="wdLxVoXEwLSg8JIdVliqp9iXQSq" width="690" height="445" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e1d3f50f2404ec5727eb485183ba830b52e43a66_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot_16.png</span><span class="informations">852×550 29.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p><strong>During the retry</strong></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/38c2a591d93d48cd3f798de75ab08208080f74c3.png" data-download-href="/uploads/short-url/867LyIXpJS5xVN37fNOKGgDaX51.png?dl=1" title="Без названия (1).png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/17360817/181097994-55d5ad48-866d-4ffb-b45c-8a81aa99298f.png" alt="Без названия (1).png" data-base62-sha1="867LyIXpJS5xVN37fNOKGgDaX51" width="690" height="314" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/38c2a591d93d48cd3f798de75ab08208080f74c3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Без названия (1).png</span><span class="informations">841×383 20.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/alexey-igrychev">@alexey-igrychev</a> ,</p>
<p>There is an earlier reported ticket for the same issue:</p>
<p><a href="https://github.community/t/github-ref-is-inconsistent/17728">https://github.community/t5/GitHub-Actions/GITHUB-REF-is-inconsistent/td-p/48129</a></p>
<p>After re-run the failed workflow for <strong>pull_request</strong> event, the  <strong>github.ref</strong> and  <strong>github.sha</strong> will be changed to the  <strong>ref</strong> and  <strong>sha</strong>  of  <strong>head </strong> (the source branch of the pull request), instead of keeping the original  <strong>ref</strong> and <strong>sha</strong>.</p>
<p>I had reported this issue to the appropriate engineering team, the team is investigating and evaluating this issue.</p>
